src/components/calcite-block/calcite-block.tsx

```diff
@@ -115,6 +115,18 @@ export class CalciteBlock {
     this.calciteBlockToggle.emit();
   };
 
+  onHeaderKeyDownOrUp = (event: KeyboardEvent) => {
+    const { currentTarget, key, target } = event;
+
+    if (key === " " || key === "Enter") {
+      const shouldBlockBubbledControlEvent = currentTarget !== target;
+
+      if (shouldBlockBubbledControlEvent) {
+        event.preventDefault();
```

Comment:
Did you mean to do event.stopPropagation()? This wouldn't stop the event from bubbling.

Reply:
That's intentional. I still want the event to propagate, but not fire a click event on the parent button element.

Reply:
Is this a situation where we have nested interactive elements? That's usually an anti-pattern. There a new way we could archictect this structure so we don't need this?

Reply:
I can't think of a way to restructure this at the moment. I could stop event propagation, but this means users would no longer be able to capture bubbled key events. Maybe that's fine for now since they can listen at the nested control level?

> Is this a situation where we have nested interactive elements? That's usually an anti-pattern. 

For my own purposes, could you share a link?

Reply:
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/button

Verify here: https://validator.w3.org/#validate_by_input. Paste the following.
```<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <meta http-equiv="X-UA-Compatible" content="ie=edge">
    <title>Document</title>
</head>
<body>
    <button>
        <input type="text" />
    </button>
</body>
</html>```